src/servers/src/http.rs

```diff
@@ -503,14 +503,19 @@ impl HttpServer {
             );
         }
 
-        // mem profiler
-        #[cfg(feature = "mem-prof")]
-        {
-            router = router.nest(
-                &format!("/{HTTP_API_VERSION}/prof"),
-                Router::new().route("/mem", routing::get(crate::http::mem_prof::mem_prof)),
-            );
-        }
+        // prof routers
+        router = router.nest(
```

Comment:
Consider adding a feature gate for this.

Reply:
Agree, and disable it by default.

Reply:
Do we need to disable it by default? This library only does samples when we trigger the profiling explicitly. This is different from the heap profiling feature that replaces the heap allocator.

So both tikv and databend add the pprof API by default and don't provide a feature to disable it.
https://github.com/tikv/tikv/blob/9aa1d7350db02900998a8dcb0f77dcc01341c169/src/server/status_server/mod.rs#L633-L635

https://github.com/datafuselabs/databend/blob/67f445e83cd4eceda98f6c1c114858929d564029/src/query/service/src/api/http_service.rs#L72

Reply:
Otherwise we will need permission check for this API to prevent it from being turned on by some random user.

Reply:
@sunng87 I get you. I will add a feature.